scripts/booktape_report.py

```
@@ -0,0 +1,146 @@
+    out.append("  `spread_capture_frac` asserts 0.25 of volume is captured; the"
+               " figures above are an UPPER BOUND on that, before queue.")
+    out.append("")
+
+    out.append("THE TOUCH PAIR -- best_bid(UP) + best_bid(DOWN)")
+    ptotal = sum(p[1] for p in pairs) or 0
+    for cost, n in pairs:
+        verdict = ("REFUSED by max_pair_cost" if cost >= MAX_PAIR_COST
+                   else "tradeable")
+        out.append(f"  ${cost:.4f}  n={n:<6} {_pct(n, ptotal)}  {verdict}")
+    refused = sum(n for c, n in pairs if c >= MAX_PAIR_COST)
+    out.append(f"  refused by the $%.2f cap: %s of samples"
```

Ruff (0.16.3)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)